tests/test_server.py

```diff
@@ -203,7 +241,58 @@ def test_server_export_remote(self):
         if httpd:
             httpd.socket.close()
 
+    def test_server_update_deployment(self):
+        with self.runner.isolated_filesystem():
+            init_project(
+                self.runner,
+                args=["init", "--mono"],
+                env=dict(UNFURL_HOME=""),
+            )
+
+            initial_deployment = deployment.format("initial")
+            # Create a mock deployment
+            os.makedirs("dashboard/deployments/dev")
+            with open ("dashboard/deployments/dev/deployment.json", "w") as f:
+                f.write(initial_deployment)
+
+            repo = Repo.init('.')
+            repo = GitRepo(repo)
+            repo.add_all()
+            repo.commit_files(["dashboard/deployments/dev/deployment.json"],"Add deployment")
+
+            p = Process(
+                target=server.serve,
+                args=("localhost", 8082, None, ".", ".", {"home": ""}),
+            )
+            assert start_server_process(p, 8082)
+
+            target_patch = patch.format("target")
+            res = requests.post(
+                "http://localhost:8082/update_deployment",
+                json={
+                    "projectPath": ".",
+                    "path": "dashboard/deployments/dev/deployment.json",
+                    "patch": json.loads(target_patch),
+                }
+            )
+
+            assert res.status_code == 200
+
+            with open ("dashboard/deployments/dev/deployment.json", "r") as f:
+                data = json.load(f)
+                assert (data['ResourceTemplate']
```

Comment:
this looks good but it would be great if we could add a test for a patch with a delete op